configure.ac

```diff
@@ -233,42 +263,38 @@ fi
 
 #-------------------------------------------------------------------------------
 
-LDFLAGS="$CFLAGS"
-LDFLAGS_BIN="$LDFLAGS"
-
-if test "x$FC" = "xnone" ; then
-  AC_MSG_NOTICE(cfitsio: == Fortran compiler search has been overridden)
-  AC_MSG_NOTICE(cfitsio: == Cfitsio will be built without Fortran wrapper support)
-  FC=
-  F77_WRAPPERS=
-else
-  AC_CHECK_PROGS(FC, gfortran g95 g77 f77 ifort f95 f90 xlf cf77 gf77 af77 ncf f2c, notfound)
-  if test $FC = 'notfound' ; then
-    AC_MSG_WARN(cfitsio: == No acceptable Fortran compiler found in \$PATH)
-    AC_MSG_NOTICE(cfitsio: == Adding wrapper support for GNU Fortran by default)
-    CFORTRANFLAGS="-Dg77Fortran"
-    F77_WRAPPERS="\${FITSIO_SRC}"
-  else
-    CFORTRANFLAGS=
-    F77_WRAPPERS="\${FITSIO_SRC}"
-    echo $ac_n "checking whether we are using GNU Fortran""... $ac_c" 1>&6
-    if test `$FC --version -c < /dev/null 2> /dev/null | grep -c GNU` -gt 0 -o \
-            `$FC --version -c < /dev/null 2> /dev/null | grep -ic egcs` -gt 0
-    then
-      echo "$ac_t""yes" 1>&6
-      echo $ac_n "cfitsio: == Adding wrapper support for GNU Fortran""... $ac_c" 1>&6
-      CFORTRANFLAGS="-Dg77Fortran"
-      echo "$ac_t"" done" 1>&6
-    else
-      echo "$ac_t""no" 1>&6
-      if test $FC = 'f2c' ; then
-        echo $ac_n "cfitsio: == Adding wrapper support for f2c""... $ac_c" 1>&6
+# Fortran wrappers:
+# -----------------
+
+# Use GNU Fortran wrappers by default:
+
+if test "x$use_fortran" = "xyes"; then
+  AC_MSG_NOTICE(cfitsio: == Adding Fortran wrappers)
+  #AC_PROG_FC
+  if test "x$FC" != "x" ; then F77=$FC; fi
+  AC_PROG_F77
+  #AC_CHECK_PROGS(FC, gfortran g95 g77 f77 ifort f95 f90 xlf cf77 gf77 af77 ncf f2c, notfound)
+  if test "x$F77" != "x" ; then
+     #FC_BASENAME=`basename $FC`
+     FC_BASENAME=`basename $F77`
+     if test "x$FC_BASENAME" = "xf2c" ; then
+        AC_MSG_NOTICE(cfitsio: == Adding wrapper support for f2c)
```

Comment:
What about using one of these built-in autoconf macros to detect the Fortran compiler? https://www.gnu.org/software/autoconf/manual/autoconf-2.67/html_node/Fortran-Compiler.html

Reply:
It is using AC_PROG_F77 - are you thinking it should be using a different macro?

Reply:
Oh, no, I just didn't notice that. I think I was looking at AC_CHECK_PROGS and didn't realize that it was commented out.